overlord/state/taskrunner.go

```diff
@@ -301,7 +342,7 @@ func (r *TaskRunner) run(t *Task) {
 				r.taskErrorCallback(err)
 			}
 		}
-
+		r.checkExhaustion()
```

Comment:
does the helper needs to check all the tasks or only t.Change()

Reply:
Only t.Change() which turns out has simplified the solution *a lot*, thank you so much for catching this!